linux-anvil/Dockerfile

```diff
@@ -46,34 +47,41 @@ RUN curl -s -L -O https://github.com/krallin/tini/releases/download/v0.10.0/tini
     chmod +x tini && \
     mv tini /usr/local/bin
 
+# Add a file for users to source to activate the `conda`
+# environment `root` and the devtoolset compiler. Also
+# add a file that wraps that for use with the `ENTRYPOINT`.
+COPY entrypoint_source /opt/docker/bin/entrypoint_source
+COPY entrypoint /opt/docker/bin/entrypoint
+
+# Create a non-root user and give sudo permission for yum
+RUN adduser conda && \
+    echo 'conda ALL=NOPASSWD: /usr/bin/yum' >> /etc/sudoers
+
+# From now on, root should not be used any more
+USER conda
+
 # Install the latest Miniconda with Python 3 and update everything.
-RUN curl -s -L http://repo.continuum.io/miniconda/Miniconda3-4.1.11-Linux-x86_64.sh > miniconda.sh && \
-    openssl md5 miniconda.sh | grep 874dbb0d3c7ec665adf7231bbb575ab2 && \
-    bash miniconda.sh -b -p /opt/conda && \
-    rm miniconda.sh && \
-    export PATH=/opt/conda/bin:$PATH && \
+RUN curl -s -L http://repo.continuum.io/miniconda/Miniconda3-4.1.11-Linux-x86_64.sh > ~/miniconda.sh && \
+    openssl md5 ~/miniconda.sh | grep 874dbb0d3c7ec665adf7231bbb575ab2 && \
+    bash ~/miniconda.sh -b -p ~/conda && \
+    rm ~/miniconda.sh && \
+    export PATH=$HOME/conda/bin:$PATH && \
```

Comment:
As mentioned in the comment below, one thing to consider is simply leaving this where it is but running `chown` on `/opt/conda` to put it in the `conda` user's hands.